### ISSUE TITLE
OwinStartupAttribute required in web.config to correct Server Error

### ISSUE THREAD

**Issue by ElimAdmin** [SparkDevNetwork/Rock#884]
The following Server Error occurred without me making any changes to the server, nor have I upgraded Rock.

This was fixed by adding 
```xml
<add key="owin:AutomaticAppStartup" value="false" /> 
```
missing from the 
```xml
<appSettings> 
```
section of web.config.

This has fixed the issue for us, but I'm logging this as an issue in case there are any follow up actions required.  http://stackoverflow.com/questions/20068075/owin-startup-class-missing

This also created a Mandrill webhook error email:
Your Mandrill webhook at URL http://elimrms.org/Webhooks/Mandrill.ashx has failed with the following error: POST to http://elimrms.org/Webhooks/Mandrill.ashx failed with 500: 
and then the following error message

------

Server Error in '/' Application.
The following errors occurred while attempting to load the app.
- No assembly found containing an OwinStartupAttribute.
- The discovered startup type 'Startup, App_Code.fwubskye, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' conflicts with the type 'Startup, App_Code.90te6aoh, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. Remove or rename one of the types, or reference the desired type directly.
To disable OWIN startup discovery, add the appSetting owin:AutomaticAppStartup with a value of "false" in your web.config.
To specify the OWIN startup Assembly, Class, or Method, add the appSetting owin:AppStartup with the fully qualified startup class or configuration method name in your web.config.
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code.

Exception Details: System.EntryPointNotFoundException: The following errors occurred while attempting to load the app.
- No assembly found containing an OwinStartupAttribute.
- The discovered startup type 'Startup, App_Code.fwubskye, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' conflicts with the type 'Startup, App_Code.90te6aoh, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. Remove or rename one of the types, or reference the desired type directly.
To disable OWIN startup discovery, add the appSetting owin:AutomaticAppStartup with a value of "false" in your web.config.
To specify the OWIN startup Assembly, Class, or Method, add the appSetting owin:AppStartup with the fully qualified startup class or configuration method name in your web.config.

Source Error:

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Stack Trace:

[EntryPointNotFoundException: The following errors occurred while attempting to load the app.
 - No assembly found containing an OwinStartupAttribute.
 - The discovered startup type 'Startup, App_Code.fwubskye, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' conflicts with the type 'Startup, App_Code.90te6aoh, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. Remove or rename one of the types, or reference the desired type directly.
To disable OWIN startup discovery, add the appSetting owin:AutomaticAppStartup with a value of "false" in your web.config.
To specify the OWIN startup Assembly, Class, or Method, add the appSetting owin:AppStartup with the fully qualified startup class or configuration method name in your web.config.]
   Microsoft.Owin.Host.SystemWeb.OwinBuilder.GetAppStartup() +357
   Microsoft.Owin.Host.SystemWeb.OwinHttpModule.InitializeBlueprint() +28
   System.Threading.LazyInitializer.EnsureInitializedCore(T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory) +86
   Microsoft.Owin.Host.SystemWeb.OwinHttpModule.Init(HttpApplication context) +106
   System.Web.HttpApplication.RegisterEventSubscriptionsWithIIS(IntPtr appContext, HttpContext context, MethodInfo[] handlers) +418
   System.Web.HttpApplication.InitSpecial(HttpApplicationState state, MethodInfo[] handlers, IntPtr appContext, HttpContext context) +172
   System.Web.HttpApplicationFactory.GetSpecialApplicationInstance(IntPtr appContext, HttpContext context) +336
   System.Web.Hosting.PipelineRuntime.InitializeApplication(IntPtr appContext) +296

[HttpException (0x80004005): The following errors occurred while attempting to load the app.
 - No assembly found containing an OwinStartupAttribute.
 - The discovered startup type 'Startup, App_Code.fwubskye, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' conflicts with the type 'Startup, App_Code.90te6aoh, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. Remove or rename one of the types, or reference the desired type directly.
To disable OWIN startup discovery, add the appSetting owin:AutomaticAppStartup with a value of "false" in your web.config.
To specify the OWIN startup Assembly, Class, or Method, add the appSetting owin:AppStartup with the fully qualified startup class or configuration method name in your web.config.]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +9930568
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +101
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +254

Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.0.30319.34209

**Comment by azturner** [comment]
@ElimAdmin , Adding config setting to stop OWIN from doing automatic startup is probably not the best solution to this issue. It looks like the problem you were experiencing is due to framework finding two Startup classes in the IIS Temp folders. If you delete the IIS temp folder ( `\Windows\Microsoft.NET\Framework(64)\v4.0.30319\Temporary ASP.NET Files\*` ) for Rock, and remove the config setting you added, that should also fix the issue.

We are making a small config change in v3.0 (https://github.com/SparkDevNetwork/Rock/commit/bfdff9fe7bde2b8dd4ba67793160f961026037c9) to specify the OWIN Startup class that should help prevent this issue the future, but being unable to reproduce the error makes it difficult to know for sure.

**Comment by treyhendon** [comment]
FWIW I experienced a problem with this as well, but because I'm on a shared account I can't delete the files and am using the workaround from @ElimAdmin until 3.0 comes out.  Thanks!

**Comment by azturner** [comment]
FYI.. v3.0 ( and 3.1 ) were released this morning.

**Comment by waseemmuhammad-vteams** [comment]
It works.

Thanks

**Comment by karim007** [comment]
thanks a lot

**Comment by aditiparakh111** [comment]
Thanks a lot ! worked perfectly for me...

**Comment by seun104** [comment]
Thanks a lot!

**Comment by shoeb-hasan** [comment]
We have webApps and PaaS on Azure, so we don't have access to IIS. The following app-settings helped us resolving the issue:
<add key="owin:AppStartup" value="YourNameSpace_and_startup_className"/>

**Comment by AshleyAS** [comment]
I cannot add <add key="owin:AutomaticAppStartup" value="false" /> to my web config because it ignores the Startup.cs and keeps hangfire from operating properly. Any other solutions?

**Comment by kmw12321** [comment]
> We have webApps and PaaS on Azure, so we don't have access to IIS. The following app-settings helped us resolving the issue:
> 
> <add key="owin:AppStartup" value="YourNameSpace_and_startup_className"/>
> 

This I think is exactly my problem. Can anyone help me out with how I would determine what my namespace and startup class name are for my Rock instance? I just migrated from a windows VM we were hosting to Azure and I started getting errors on trying to load the site. I got it to load by adding "False" instead of what you provided above, but login was not possible after that.
